pyproject.toml

```diff
@@ -1,10 +1,114 @@
-# build-system
-# - ref 1: https://peps.python.org/pep-0517/
-#
 [build-system]
-requires = ["jupyter_packaging~=0.7.9", "jupyterlab~=3.0", "setuptools>=40.8.0", "wheel"]
-build-backend = "setuptools.build_meta"
+build-backend = "hatchling.build"
+requires = [
+    "hatch-jupyter-builder >=0.5",
+    "hatch-nodejs-version",
+    "hatchling >=1.4.0",
+    "jupyterlab >=3.4.7,<4.0.0",
+]
+
+[project]
+name = "jupyter_server_proxy"
+dynamic = [
+    "authors",
+    "description",
+    "keywords",
+    "urls",
+    "version",
+]
+readme = "README.md"
+license = { file = "LICENSE" }
+requires-python = ">=3.7"
+classifiers = [
+    "Framework :: Jupyter",
+    "Framework :: Jupyter :: JupyterLab",
+    "Framework :: Jupyter :: JupyterLab :: 2",
+    "Framework :: Jupyter :: JupyterLab :: 3",
+    "Framework :: Jupyter :: JupyterLab :: Extensions :: Prebuilt",
+    "Framework :: Jupyter :: JupyterLab :: Extensions",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: MacOS :: MacOS X",
+    "Operating System :: Microsoft :: Windows",
+    "Operating System :: POSIX :: Linux",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+dependencies = [
+    "aiohttp",
+    "jupyter-server >=1.0",
+    "simpervisor >=0.4",
+]
+
+[project.optional-dependencies]
+test = [
+    "pytest",
+    "pytest-cov",
+    "pytest-html",
+]
+acceptance = [
+    "jupyter-server-proxy[test]",
+    "notebook <7",
+    "robotframework-jupyterlibrary >=0.4.2",
+]
+docs = [
+    "myst-parser",
+    "sphinx-autobuild",
+    "sphinx-book-theme",
+    "sphinx-copybutton",
+    "sphinxext-opengraph",
+    "sphinxext-rediraffe",
+]
```

Comment:
I did a similar change to introduce `docs` as a optional-dependencies target in another project in the jupyterhub org, and we agreed to just stick with the requirements.txt under docs/ so I figure we do this also in this project.

If RTD needs nodejs btw, it can be installed via configuration for us in the v2 version of their .readthedocs.yaml config file.

Reply:
```suggestion
```